internal/receiver/smartagentreceiver/receiver.go

```diff
@@ -135,5 +148,79 @@ func (r *Receiver) createMonitor(monitorType string) (interface{}, error) {
 		output.AddExtraDimension(k, v)
 	}
 
-	return monitor, nil
+	// Configure SmartAgentConfigProvider to gather any global config overrides and
+	// set required envs.
+	configureEnvironmentOnce.Do(func() {
+		r.setUpSmartConfigProvider(extensions)
+		r.setUpEnvironment()
+	})
+
+	// Note, that this receiver has to configure main collectd even for collectd/custom monitor,
+	// despite the fact that, that monitor stands up its own instance of collectd to prevent this
+	// panic "Main collectd instance should not be accessed before being configured".
+	if r.config.monitorConfig.MonitorConfigCore().IsCollectdBased() {
+		configureCollectdOnce.Do(func() {
+			r.logger.Info("Configuring collectd")
+			err = configureMainCollectd(r.getCollectdConfig())
+		})
+	}
+
+	return monitor, err
+}
+
+func configureMainCollectd(collectdConfig *config.CollectdConfig) error {
+	return collectd.ConfigureMainCollectd(collectdConfig)
+}
+
+func (r *Receiver) setUpSmartConfigProvider(extensions map[configmodels.Extension]component.ServiceExtension) {
```

Comment:
would prefer `setUpSmartAgentConfigProvider` or `setUpGlobalConfigProvider`

Reply:
Fixed, that was not intentional.